libgalois/src/PropertyGraph.cpp

```diff
@@ -1289,3 +1306,16 @@ katana::PropertyGraph::GetNodePropertyIndex(
   }
   return KATANA_ERROR(katana::ErrorCode::NotFound, "node index not found");
 }
+
+katana::Result<void>
+katana::PropertyGraph::RecreatePropertyIndexes() {
+  for (const std::string& column_name : rdg_.node_property_index_columns()) {
+    KATANA_CHECKED(MakeNodeIndex(column_name));
+  }
```

Comment:
Currently a request to make an index only looks at the PropertyGraph, so will fail unless the property is loaded into memory. I had intended to defer any questions of in/out-of-core properties, however it would be simple here to only try to make the index if the column is loaded (otherwise this will fail the entire graph load) - I'll make that change.

There will still exist the bugaboo which sounds like something you were describing a while ago - we load indexes only for in-core properties, then only persist those, losing indexes for out-of-core properties. That I would like to defer for this change - having a basic index flow in place, as well as me understanding more, will help do that correctly in the future.

Reply:
Yes, the in-core/out-of-core logic does get tricky.

I like your plan--only make the index if the column is loaded.  

For the grand and glorious future, you can look at `RDGPartHeader.h` for the state machine for properties.  We also have `full_node|edge_schema` and` loaded_node|edge_schema`, so you do have access to all properties all the time, even if they aren't loaded.